rust/bindgen/Cargo.toml

```diff
@@ -11,10 +11,10 @@ crate-type = ['cdylib', 'rlib']
 [dependencies]
 anyhow = "1.0"
 heck = { version = "0.4", features = ["unicode"] }
-wit-component = "0.9"
-wit-parser = "0.7"
+wit-parser = { git = "https://github.com/bytecodealliance/wasm-tools.git", rev="4fdfc3869b988bfcdb7660f61b38b685f8389d94"}
+wit-component = { git = "https://github.com/bytecodealliance/wasm-tools.git", rev="4fdfc3869b988bfcdb7660f61b38b685f8389d94"}
```

Comment:
Oh are there changes that have yet to be published which these depend on?

Reply:
Yeah I think so. I'll remove them and show the issue that it reports (won't be until tomorrow though).

Reply:
I'm seeing the following compilation errors when using the lastest available versions of `wit-parser` (0.8.0) and `wit-component` (0.11.0) on crates.io:
```console
cargo build --release --target=wasm32-wasi -p=bindgen
   Compiling pulldown-cmark v0.8.0
   Compiling wasm-metadata v0.8.0
   Compiling wit-parser v0.8.0
   Compiling wit-component v0.11.0
   Compiling bindgen v0.0.0 (/home/danielbevenius/work/wasm/wasmtime-py/rust/bindgen)
error[E0599]: no variant or associated item named `Resource` found for enum `wit_parser::TypeDefKind` in the current scope
    --> bindgen/src/bindgen.rs:1003:34
     |
1003 |                     TypeDefKind::Resource => unreachable!(),
     |                                  ^^^^^^^^ variant or associated item not found in `TypeDefKind`

error[E0599]: no variant or associated item named `Handle` found for enum `wit_parser::TypeDefKind` in the current scope
    --> bindgen/src/bindgen.rs:1004:34
     |
1004 |                     TypeDefKind::Handle(_) => unreachable!(),
     |                                  ^^^^^^ variant or associated item not found in `TypeDefKind`

error[E0599]: no variant or associated item named `Resource` found for enum `wit_parser::TypeDefKind` in the current scope
    --> bindgen/src/bindgen.rs:1155:30
     |
1155 |                 TypeDefKind::Resource => todo!("generate for resource"),
     |                              ^^^^^^^^ variant or associated item not found in `TypeDefKind`

error[E0599]: no variant or associated item named `Handle` found for enum `wit_parser::TypeDefKind` in the current scope
    --> bindgen/src/bindgen.rs:1156:30
     |
1156 |                 TypeDefKind::Handle(_) => todo!("generate for handle"),
     |                              ^^^^^^ variant or associated item not found in `TypeDefKind`

error[E0599]: no variant or associated item named `Method` found for enum `wit_parser::FunctionKind` in the current scope
    --> bindgen/src/bindgen.rs:2519:35
     |
2519 |                     FunctionKind::Method(_)
     |                                   ^^^^^^ variant or associated item not found in `FunctionKind`

error[E0599]: no variant or associated item named `Static` found for enum `wit_parser::FunctionKind` in the current scope
    --> bindgen/src/bindgen.rs:2520:37
     |
2520 |                     | FunctionKind::Static(_)
     |                                     ^^^^^^ variant or associated item not found in `FunctionKind`

error[E0599]: no variant or associated item named `Constructor` found for enum `wit_parser::FunctionKind` in the current scope
    --> bindgen/src/bindgen.rs:2521:37
     |
2521 |                     | FunctionKind::Constructor(_) => todo!(),
     |                                     ^^^^^^^^^^^ variant or associated item not found in `FunctionKind`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `bindgen` (lib) due to 7 previous errors
make: *** [Makefile:3: bindgen] Error 101

```

Reply:
Ended up being a bit busy today but I'll dig in to this tomorrow and see if I can figure out what's going on.

Reply:
Ah ok I see what's going on here. I think it would be good to use the crates.io-published versions of these crates rather than git dependencies. Support for resources isn't required for this repository at this time, so all of the above errors can be fixed by removing the branches of the `match` statements.

Reply:
@alexcrichton I've updated this now. 

Please let me know if these changes look alright and I'll squash the commits.